packages/backend/app/Http/Controllers/FavoriteController.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Favorite;
+use App\Models\Message;
+use App\OpenApi\RequestBodies\FavoriteReverseRequestBody;
+use App\OpenApi\Responses\BadRequestResponse;
+use App\OpenApi\Responses\FavoriteReverseResponse;
+use App\OpenApi\Responses\UnauthorizedRequestResponse;
+use Exception;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Contracts\Routing\ResponseFactory;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Validator;
+use Ramsey\Uuid\Uuid;
+use Vyuldashev\LaravelOpenApi\Attributes as OpenApi;
+
+#[OpenApi\PathItem]
+class FavoriteController extends Controller
+{
+    /**
+     * お気に入り登録用エンドポイント
+     *
+     * @param  Request  $request
+     * @param  string  $messageId
+     * @return Response|Application|ResponseFactory
+     */
+    #[OpenApi\Operation]
+    #[OpenApi\RequestBody(factory: FavoriteReverseRequestBody::class)]
+    #[OpenApi\Response(factory: FavoriteReverseResponse::class)]
+    #[OpenApi\Response(factory: BadRequestResponse::class)]
+    #[OpenApi\Response(factory: UnauthorizedRequestResponse::class)]
+    public function addFavorite(Request $request, string $messageId): Response|Application|ResponseFactory
+    {
+        $validator = Validator::make($request->all(), [
+            'isFavorite' => ['required', 'boolean'],
+        ]);
+
+        if ($validator->fails()) {
+            return response(null, 400);
+        }
+
+        if ($request->user() === null) {
+            return response(null, 401);
+        }
+
+//        $messageId = $request->route('messageId');
+
+        $message = Message::find($messageId);
+        if ($message === null) {
+            return response([
+                'message' => 'Bad Request',
+            ], 400);
```

Comment:
対象のメッセージが存在しない場合にここに入ります。
404が適切なのでしょうか？
```suggestion
            return response([
                'message' => Message not found',
            ], 404);
```